src/main/webapp/blobstoreFunctions.js

```diff
@@ -0,0 +1,54 @@
+/** Loads photo */
+function loadScripts() {
+  fetchBlobstoreUrl();
+  populatePosts();
+}
+
+
+/** Loads upload image URL and attaches it to the form */
+function fetchBlobstoreUrl() {
+  fetch('/blobstore-upload-url') 
+      .then((response) => {
+        return response.text();
+      })
+      .then((imageUploadUrl) => {
+        const messageForm = document.getElementById('plant-pic-form');
+        messageForm.action = imageUploadUrl;
+      });
+}
+
+
+/**
+  * Creates and adds a photo 
+  * pic is the URL of image to be shown in the post
+*/
+function createListElement(pic) {
+  let card = document.createElement("div");
+  card.className = "card m-1";
+
+  let body = document.createElement("div");
+  body.className = "card-body";
```

Comment:
Same comment as above to avoid making so many variables

Reply:
These ones are each used twice, not sure if there's a way to consolidate creating the div, adding the class name and appending them together

Reply:
I what you mean, okay ignore this comment then lol

Reply:
I guess one thing I will just leave here is that maybe you could fix the spacing. I think there is a lot of spaces when some of the code could really be stacked on top of each other since it all goes together if that makes sense?

Reply:
yep, I'll fix the spacing!